module/signature/aggregation_test.go

```diff
@@ -218,6 +230,34 @@ func TestAggregatorSameMessage(t *testing.T) {
 		}
 	})
 
+	t.Run("identity aggregated signature", func(t *testing.T) {
```

Comment:
I like how you describe the different failure cases here: https://github.com/onflow/flow-go/blob/35738f2618fb9dd87060955fe5bef0b232341992/module/signature/aggregation_test.go#L191-L195

but I have trouble matching the test cases [`invalid signature`](https://github.com/onflow/flow-go/blob/35738f2618fb9dd87060955fe5bef0b232341992/module/signature/aggregation_test.go#L196) and [`identity aggregated signature`](https://github.com/onflow/flow-go/blob/35738f2618fb9dd87060955fe5bef0b232341992/module/signature/aggregation_test.go#L233) to the description

_Suggestion_
* My understanding is that failure case 1 and 3 are what is tested in [`invalid signature`](https://github.com/onflow/flow-go/blob/35738f2618fb9dd87060955fe5bef0b232341992/module/signature/aggregation_test.go#L196) (?) If that is the case, maybe we can revise the description to focus on the expected sentinel. For test  [`invalid signature`](https://github.com/onflow/flow-go/blob/35738f2618fb9dd87060955fe5bef0b232341992/module/signature/aggregation_test.go#L196), we could revise test [`invalid signature`](https://github.com/onflow/flow-go/blob/35738f2618fb9dd87060955fe5bef0b232341992/module/signature/aggregation_test.go#L190-L199) to 
    ```golang
    // There are two scenarios, where `Aggregate()` is expected to return an `InvalidSignatureIncludedError`:
    //  1. Some signature(s) added via `TrustedAdd` has an invalid structure, i.e. it can't be deserialized
    //     successfully. In this case, already the aggregation step fails.
    //  2. The signatures were deserialized successfully, but the aggregate signature doesn't verify to the
    //     aggregate public key (although it is not identity).
    t.Run("invalid signature", func(t *testing.T) {
      _, s, _ := createAggregationData(t, 1)
      invalidStructureSig := (crypto.Signature)([]byte{0, 0}) // case 1: signature with invalid structure
      mismatchingSig := s[0]                                  // case 2: structurally valid sig, but doesn't match respective key
         ⋮
    ```
* For test [`identity aggregated signature`](https://github.com/onflow/flow-go/blob/35738f2618fb9dd87060955fe5bef0b232341992/module/signature/aggregation_test.go#L233):
   * would suggest to update the documentation 
      ```golang
      // `Aggregate()` is expected to return an `ErrIdentityPublicKey` when the aggregated signature is identity signature;
      //  even if the signatures are all individually valid
      t.Run("identity aggregated signature", func(t *testing.T) {
         ⋮
      ```
   * Same recommendation as in [my comment above](): minimal test cases without any overlap with the [`invalid signature` test case](https://github.com/onflow/flow-go/blob/35738f2618fb9dd87060955fe5bef0b232341992/module/signature/aggregation_test.go#L196); in other words, lets make sure `ErrIdentityPublicKey` is returned even if all signatures are individually valid.

Reply:
> I like how you describe the different failure cases here

I think this was your description from your suggestion PR in the previous iteration :) 
I have updated the comment because we shifted the logic from checking for identity signatures to checking for identity public key. Please refer to this comment for more details : https://github.com/onflow/flow-go/pull/3441#discussion_r1095435637